flink-connectors/flink-connector-base/src/test/java/org/apache/flink/connector/base/source/hybrid/HybridSourceSplitEnumeratorTest.java

```diff
@@ -177,6 +177,26 @@ public void testRestoreEnumerator() throws Exception {
                 .hasSize(1);
     }
 
+    @Test
+    public void testRestoreEnumeratorWith2ndSource() throws Exception {
```

Comment:
Let's try to find a name for this test case that better reflects the issue? What we are testing is restore with second or subsequent source after the first source did not assign any splits? The latter is key to the issue, because the reader does not (need to) track the last seen source index. How about `testRestoreEnumeratorAfterFirstSourceWithoutRestoredSplits`?

Reply:
Yes, you are right